src/gsAssembler/gsRemapInterface.h

```diff
@@ -1,132 +1,185 @@
+  * is an affine-linear function. In this case, this class uses that affine-linear function.
+  *
+  * If the mapping is found out not to be affine-linear, this class constructs an approximation of the interface
+  * map. This is available only if \f$ \Omega_1 \f$ and \f$ \Omega_2 \f$ are two dimensional, i.e., the interface
+  * itself is one dimensional. We refer to this as the 2D case.
+  *
+  * @ingroup Assembler
+ **/
 template <class T>
 class gsRemapInterface : public gsFunction<T>
 {
 public:
+
     /// Shared pointer for gsRemapInterface
     typedef memory::shared_ptr< gsRemapInterface > Ptr;
 
     /// Unique pointer for gsRemapInterface
     typedef memory::unique_ptr< gsRemapInterface > uPtr;
 
-    /// Unique pointer to domain iterator
-    typedef memory::unique_ptr< gsDomainIterator<T> > domainIterUPtr;
+public:
 
-    /// Constructor which takes a multipatch and a boundary interface, useful if the interface is fully matching
-    gsRemapInterface(const gsMultiPatch<T> & mp, const gsMultiBasis<T> & basis, const boundaryInterface & bi);
+    enum {
+        neverAffine = -1,  ///< Instructs constructor not to use an affine mapping
+        alwaysAffine = 0   ///< Instructs constructor always to use an affine mapping
+    };
 
-    /// Constructor for the class which takes two geometries
-    gsRemapInterface(const gsGeometry<T> & g1, const gsGeometry<T> & g2, const gsBasis<T>* basis1, const gsBasis<T>* basis2) : m_g1(g1), m_g2(g2), m_b1(*basis1), m_b2(*basis2)
-    {
-        GISMO_ASSERT(m_g1.geoDim() == m_g2.geoDim(), "The two given geometries do not have the same dimension!");
-        findInterface();
-        constructReparam();
-        if(!m_isMatching)
-            constructBreaks();
-   }
+    /// @brief Constructor
+    ///
+    /// @param  mp                 The multi-patch object
+    /// @param  mb                 The multi-basis object
+    /// @param  bi                 The boundary interface (specifying the patches and their \a patchSide s)
+    /// @param  checkAffine        The number of interior points (per direction) in point grid used to
+    ///                            check if the mapping is affine. If set to i, the grid consits of i interor
+    ///                            point and the two boundary points per direction, so \f$ (i+2)^d \f$ points.
+    ///                            For \a alwaysAffine or \a notAffine, no checks are performed.
+    /// @param  numSamplePoints    Number of sample points for computing fitting curve (if not affine)
+    /// @param  intervalsOfFittingCurve  Number of knot space in the fitting curve (if not affine)
+    /// @param  degreeOfFittingCurve     Spline degree of fitting curve (if not affine)
+    /// @param  equalityTolerance  Points are considered equal iff difference is smaller.
+    /// @param  newtonTolerance    Tolerance for Newton solvers (should be significantly smaller than
+    ///                            equalityTolerance)
```

Comment:
would be better to have a gsOptionList eg as argument here to avoid long list of parameters, or to anticipate adding/changing the parameters in the future.
The documentation can then refer to the expected options..